src/shared/tpm2-util.c

```diff
@@ -717,9 +739,26 @@ static int tpm2_make_pcr_session(
         return r;
 }
 
+static void *hash_pin(const char *pin, uint8_t ret_digest[static SHA256_DIGEST_SIZE]) {
+        struct sha256_ctx hash;
+        size_t pin_len;
+
+        assert(pin);
+        pin_len = strlen(pin);
+
+        sha256_init_ctx(&hash);
+        sha256_process_bytes(pin, pin_len, &hash);
+        sha256_finish_ctx(&hash, ret_digest);
+
+        explicit_bzero_safe(&hash, sizeof(hash));
+
+        return ret_digest;
```

Comment:
There's no point in this function having a return value, it cannot fail.

Reply:
Not even sure why I did that...